src/main/Video.tsx

```diff
@@ -196,6 +222,23 @@ const VideoPlayer: React.FC<{dataKey: number, url: string, isPrimary: boolean}>
     file: { attributes: { tabIndex: '-1' }}
   }
 
+  // External functions
+  useImperativeHandle(forwardRefThing, () => ({
+    // Renders the current frame in the video element to a canvas
+    // Returns the data url
+    captureVideo() {
+      const video = ref.current?.getInternalPlayer() as HTMLVideoElement
+      var canvas = document.createElement("canvas");
+      canvas.width = video.videoWidth;
+      canvas.height = video.videoHeight;
+      var canvasContext = canvas.getContext("2d");
+      if (canvasContext !== null) {
+        canvasContext.drawImage(video, 0, 0);
+        return canvas.toDataURL('image/png')
```

Comment:
If I click on `Generate`, I get the exception below, and nothing happens. I am on Fedora 35 with Firefox Version 103.0.
```
Uncaught DOMException: The operation is insecure. main.30cb8141.js:2
    captureVideo Video.tsx:237
    generate Thumbnail.tsx:26
    onClick Thumbnail.tsx:256
    React 11
    bind_applyFunctionN self-hosted:1355
    Qt self-hosted:1318
    unstable_runWithPriority scheduler.production.min.js:18
    React 3
    bind_applyFunctionN self-hosted:1355
    $t self-hosted:1318
```

Reply:
That should be fixed now